userspace/engine/rule_loader.h

```diff
@@ -451,4 +453,38 @@ namespace rule_loader
 		bool warn_evttypes;
 		bool skip_if_unknown_filter;
 	};
+
+	/*!
+		\brief Represents infos about a rule update (append or replace) request
+	*/
+
+	struct rule_update_info
```

Comment:
Note for the reviewer: while we could add `std::optional`s to the existing structures, those are also used to evaluate and handle actual rules which must have those fields defined. I wouldn't want the rest of the engine to care about whether a mandatory field was defined or not (as it must be) so it sounds ok to keep those fields here.